### PR TITLE
Added working openapi specification

### LINKED ISSUE #336: Rest API: Most important API for digital shadow, sends asset specific data
This issue was split out of the #313 issues and is for the digital shadow issue in general #297.

**Functionality**
This function returns all uniqueProducts for that specific asset in the given timerange. One Datapoint contains one childUID, all parentUID's and all available alternativeUniqueProductID's. All unqiueProductTags and uniqueProductTagStrings (value and timestamp for each uniqueProduct) for the childUID are returned in the same Datapoint.

`get /{customer}/{location}/{asset}/uniqueProductsWithTags`
from `<timestamp1>` to `<timestamp2>` (in RFC 3999 Format)

**Open Questions**
- What happens if there is no productTag for that uniqueProduct? (probably null). 
- What happens if there are multiple values for one uniqueProductTag (error?)

**Example**
(Two data points. The data for one single part at one asset is combined together to one data point.)
```.js
{
	"columnNames": [
		"ValveHeadAlternativeUniqueID",
		"TorqueScrew1",
		"TorqueScrew2",
		"TorqueScrew3",
		"TorqueScrew4",
		"gasketUID",
		"childUID"
		"Timestamp_ValveHeadID"
		"Timestamp_GasketImageID"
	]
	"datapoints": [
		[
			"124000003941",
			5.2,
			5.4,
			4.9,
			5.4,
			"345g48t34ru2,
			"2502934850239587"
			1239481702
			2314092348
		]
			"124000003942",
			5.1,
			5.1,
			4.8,
			5.9,
			"2493trf8434r",
			"32ß4529ß8359"
			13204982743
			12359873404
		]
	]
}
```

### PR DESCRIPTION
**Pull request by JeremyTheocharis**
Fixes #336

Co-authored-by: Lukas Hennies <45569834+Gornoka@users.noreply.github.com>
Co-authored-by: felixmhofmann <f.hofmann@oliva-advisory.de>

### CODE REVIEW

**Comment by JeremyTheocharis** [comment]
Todo: test it

**Comment by Gornoka** [comment]
added files necessary for testing in pr below 
https://github.com/united-manufacturing-hub/united-manufacturing-hub/pull/466
see recommended changes in  #466  (pr targets this branch)
tldr:
- minor changes on api and tested generation
  generation was succesfull (enough) 
  the heavy use of anyOf in the message structure will induce some problems for ceratain generators.

- added linux generator scripts

**Review by Gornoka — CHANGES_REQUESTED** [review]
see changes in #466

**Review by Gornoka — APPROVED** [review]

**Comment by Gornoka** [comment]
maybe the index.htmls should be excluded from sonar cloud as they are not manually created, but i do not know / have the rights to do that

**Review by Gornoka — COMMENTED** [review]
.
